pennylane/workflow/_capture_qnode.py

```diff
@@ -281,11 +280,33 @@ def _make_zero(tan, arg):
     return jax.lax.zeros_like_array(arg) if isinstance(tan, ad.Zero) else tan
 
 
-def _qnode_jvp(args, tangents, **impl_kwargs):
+def _backprop(args, tangents, **impl_kwargs):
     tangents = tuple(map(_make_zero, tangents, args))
     return jax.jvp(partial(qnode_prim.impl, **impl_kwargs), args, tangents)
 
 
+diff_method_map = {"backprop": _backprop}
+
+
+def _resolve_diff_method(diff_method: str, device) -> str:
+    # check if best is backprop
+    if diff_method == "best":
+        config = qml.devices.ExecutionConfig(gradient_method=diff_method, interface="jax")
+        diff_method = device.setup_execution_config(config).gradient_method
+
+    if diff_method not in diff_method_map:
+        raise NotImplementedError(f"diff_method {diff_method} not yet implemented.")
+
+    return diff_method
+
+
+def _qnode_jvp(args, tangents, *, qnode_kwargs, device, **impl_kwargs):
+    diff_method = _resolve_diff_method(qnode_kwargs["diff_method"], device)
+    return diff_method_map[diff_method](
+        args, tangents, qnode_kwargs=qnode_kwargs, device=device, **impl_kwargs
+    )
+
+
```

Comment:
Just a note that while this is awesome, we will eventually have to reorganize where the jvp rule for different diff methods live and figure out a larger abstraction for hiding interface and gradient related inner details in a similar manner to how we do in the `interfaces` module for capture disabled execution.

Reply:
Definitely.  But I'd like to flesh things out a little bit more with some other diff methods so I know what I'm working with.

Reply:
Want me to leave a comment?

Reply:
Sure, thanks!